storage/innobase/handler/ha_innodb.cc

```diff
@@ -10994,18 +10994,44 @@ create_table_info_t::create_table_def()
 			dict_mem_table_free(table);
 		} else if (m_create_info->encrypt_type.length > 0
 			   && !Encryption::is_none(
-				   m_create_info->encrypt_type.str)) {
+				   m_create_info->encrypt_type.str)
+			   && !srv_tmp_tablespace_encrypt) {
```

Comment:
Two loads in a row of srv_tmp_tablespace_encrypt can race with changing the variable value.